File: packman/src/main/java/fkn/dlaskina/packman/element/Stone.java
package fkn.dlaskina.packman.element;

import java.awt.*;

/**
 * Definition of the Surprise class
 * @author VLaskin
 * @since 26.03.2016.
 */
public class Stone extends Elemental {

    private static final Color FILL_COLOR = new Color(125, 128, 128);

    public Stone() {
        super(ElementalType.Stone);
    }

    @Override
    public void paint(Graphics gr, Rectangle rect, final int frame) {
        gr.setColor(FILL_COLOR);
        gr.fillRect(rect.x, rect.y, rect.width, rect.height);
    }
}

File: packman/src/main/java/fkn/dlaskina/packman/element/Stone.kt
package fkn.dlaskina.packman.element

import java.awt.Color
import java.awt.Graphics
import java.awt.Rectangle

/**
 * Definition of the Surprise class
 * @author VLaskin
 * @since 26.03.2016.
 */
class Stone : Elemental(ElementalType.Stone) {

    companion object {
        private val FILL_COLOR = Color(125, 128, 128)
    }

    override fun paint(gr: Graphics, rect: Rectangle, frame: Int) {
        gr.color = FILL_COLOR
        gr.fillRect(rect.x, rect.y, rect.width, rect.height)
    }
}